src/libraries/System.Private.CoreLib/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Windows.cs

```diff
@@ -108,18 +108,33 @@ private static unsafe SafeFileHandle CreateFile(string fullPath, FileMode mode,
 
         private static unsafe void Preallocate(string fullPath, long preallocationSize, SafeFileHandle fileHandle)
         {
-            // preallocationSize must be ignored for non-seekable files, unsupported file systems
-            // and other failures other than ERROR_DISK_FULL and ERROR_FILE_TOO_LARGE
-            if (!FileStreamHelpers.TrySetFileLength(fileHandle, preallocationSize, out int errorCode)
-                && errorCode == Interop.Errors.ERROR_DISK_FULL || errorCode == Interop.Errors.ERROR_FILE_TOO_LARGE)
-            {
-                // Windows does not modify the file size if the request can't be satisfied in atomic way.
-                // posix_fallocate (Unix) implementation might consume all available disk space and fail after that.
-                // To ensure that the behaviour is the same for every OS (no incomplete or empty file), we close the handle and delete the file.
+            var allocationInfo = new Interop.Kernel32.FILE_ALLOCATION_INFO
+            {
+                AllocationSize = preallocationSize
+            };
+
+            if (!Interop.Kernel32.SetFileInformationByHandle(
+                fileHandle,
+                Interop.Kernel32.FileAllocationInfo,
+                &allocationInfo,
+                (uint)sizeof(Interop.Kernel32.FILE_ALLOCATION_INFO)))
+            {
+                int errorCode = Marshal.GetLastPInvokeError();
+
+                // we try to mimic the atomic NtCreateFile here:
```

Comment:
See my comment about the comment :)

Reply:
As per this comment https://github.com/dotnet/runtime/pull/58726#discussion_r707377357 this should be:
```suggestion
                // we try to mimic Unix behavior:
```